test/module/vendor/rocksdb_test.cpp

```diff
@@ -0,0 +1,269 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <rocksdb/db.h>
+
+#include <gtest/gtest.h>
+#include <boost/filesystem.hpp>
+
+#include "ametsuchi/impl/rocksdb_common.hpp"
+
+namespace fs = boost::filesystem;
+using namespace iroha::ametsuchi;
+
+class RocksDBTest : public ::testing::Test {
+ public:
+  void SetUp() override {
+    db_name_ = (fs::temp_directory_path() / fs::unique_path()).string();
+    db_port_ = std::make_shared<RocksDBPort>();
+    db_port_->initialize(db_name_);
+    tx_context_ = std::make_shared<RocksDBContext>();
+    db_port_->prepareTransaction(*tx_context_);
+
+    insertDb(key1_, value1_);
+    insertDb(key2_, value2_);
+    insertDb(key3_, value3_);
+    insertDb(key4_, value4_);
+    insertDb(key5_, value5_);
+  }
+
+  void TearDown() override {
+    tx_context_.reset();
+    db_port_.reset();
+    fs::remove_all(db_name_);
+  }
+
+  void insertDb(std::string_view key, std::string_view value) {
+    RocksDbCommon common(tx_context_);
+    common.valueBuffer() = value;
+    common.put(key);
+  }
+
+  std::string_view readDb(std::string_view key) {
+    RocksDbCommon common(tx_context_);
+    common.get(key);
+    return common.valueBuffer();
+  }
+
+  std::string db_name_;
+  std::shared_ptr<RocksDBPort> db_port_;
+  std::shared_ptr<RocksDBContext> tx_context_;
+
+  std::string const key1_ = "keY";
+  std::string const key2_ = "keYY";
+  std::string const key3_ = "ke1Y";
+  std::string const key4_ = "keyY";
+  std::string const key5_ = "ke";
+
+  std::string const value1_ = "vaLUe";
+  std::string const value2_ = "vaLUe2";
+  std::string const value3_ = "vaLUe3";
+  std::string const value4_ = "vaLUe4";
+  std::string const value5_ = "vaLUe5";
+};
+
+TEST_F(RocksDBTest, SimpleOperation) {
+  ASSERT_TRUE(readDb(key1_) == value1_);
+  ASSERT_TRUE(readDb(key2_) == value2_);
+  ASSERT_TRUE(readDb(key3_) == value3_);
+  ASSERT_TRUE(readDb(key4_) == value4_);
+  ASSERT_TRUE(readDb(key5_) == value5_);
+}
+
+TEST_F(RocksDBTest, SimpleDelete) {
+  RocksDbCommon common(tx_context_);
+  common.del(key3_);
```

Comment:
Should it throw if key does not exist - "cannot delete key with does not exist"

Reply:
i'm not sure i understand you

Reply:
What if element I request to erase does not exist. User must be able to know if the element(s) was erased, or there is nothing to erase

Examples:
1. std::erase returns number of elements were erased.
2. rm returns with exit code error if file does not exist

Reply:
```del``` return ```status```.

Reply:
Please add ASSERT_TRUE. And ASSERT_FALSE for 2 cases